cli/command/context/show.go

```diff
@@ -25,10 +25,10 @@ func newShowCommand(dockerCli command.Cli) *cobra.Command {
 
 func runShow(dockerCli command.Cli) error {
 	context := dockerCli.CurrentContext()
-	metadata, err := dockerCli.ContextStore().GetMetadata(context)
+	fmt.Fprintln(dockerCli.Out(), context)
+	_, err := dockerCli.ContextStore().GetMetadata(context)
 	if err != nil {
 		return err
 	}
-	fmt.Fprintln(dockerCli.Out(), metadata.Name)
 	return nil
 }
```

Comment:
As this is a new comment, we can still change; do we think we should produce an error _at all_ if the context wasn't found, or just "show the name of the context" (basically: this is what's _configured_ to use, but you'll learn if it works once you use it)?

We could still opt for printing the error (on STDERR) as a warning, but make the command always return a zero exit code ("success").

Reply:
It seems like it should mimic existing behavior for other commands.  e.g. for `inspect`:
```
if dockerCli.CurrentContext() == "" {
	return errors.New("no context specified")
}
```

Reply:
Hm... right; so `docker context inspect` expects a positional argument. It looks like there's a bug in the `compose-cli` wrapper, as it hijacks the `docker context` subcommands, and it looks like the usage information is incorrect;

This is the output for compose-cli;

```bash
docker context inspect --help
Display detailed information on one or more contexts

Usage:
  docker context inspect [flags]

Flags:
  -f, --format string   Format the output using the given Go template
  -h, --help            Help for inspect
```

And the output for the actual docker cli;

```bash
com.docker.cli context inspect --help

Usage:  docker context inspect [OPTIONS] [CONTEXT] [CONTEXT...]

Display detailed information on one or more contexts

Options:
  -f, --format string   Format the output using the given Go template
```

Reply:
As to the `docker context show`; that command was added in https://github.com/docker/cli/pull/3567. Basically its sole purpose is to show what context is _configured_, so that it can be used in prompts. An earlier PR (https://github.com/docker/cli/pull/2500) added it to `docker info` for that purpose, but `docker context show` is more convenient.

As it (likely) will be used for prompts, the question is, should it always validate? Without validation, the _value_ is still valid (it shows what's configured), but even if we check if the context is valid (can be found, and decoded), it may still be pointing to a host that's not reachable.